packages/react-web/src/inbox/WebConversationProvider.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  IConversationProvider,
+  EnvironmentConfig,
+  Conversation,
+  profileId,
+  FetchConversationsRequest,
+  Participant,
+  EnableConversationsResult,
+  GetAllConversationsResult,
+  ConversationsDisabledError,
+  WalletEntity,
+} from '@lens-protocol/react';
+import { failure, success } from '@lens-protocol/shared-kernel';
+import { IStorage } from '@lens-protocol/storage';
+import { Client, Conversation as XmtpConversation } from '@xmtp/xmtp-js';
+
+import { SignerAdapter } from './SignerAdapter';
+import { extractPeerProfileId } from './helpers';
+
+function getXmtpEnvironment(name: EnvironmentConfig['name']) {
+  switch (name) {
+    case 'production':
+      return 'production';
+    case 'development':
+      return 'dev';
+  }
+}
```

Comment:
I know you got inspired by token-gated content for this.... but it was a sloppy shortcut I took... better do proper dependency inversion of the config and feed it from the "above". Just last week I had to change token-gated to receive config from the main `environment` config cause it was getting in the way of some integration tests we are conducting. 🤷